### DISCUSSION TITLE
Organizing classes suggestion

### DISCUSSION THREAD

**Discussion by squivix** [Benji377/OpenMusic#117] [Ideas]
I am now noticing that the MainActivity, and PlayerFragment classes are getting bigger and messier now that I've added more inner classes (the adapters and  view holders for the ViewPager). I think it's about time we organize things a little bit more. So I propose the following:

- We separate adapters/viewholders/other inner classes from their Activity/Fragment and put them into their own outer class
- We put all activities, fragments, and adapter classes into a package (ui)
- We organize them in sub-packages inside the ui package. There are two ways I have in mind:
     - 3 packages by type: activates, fragments, adapters
     - 2 packages by place: main, player. Then all main adapters/viewholders/broadcast receiver  will be in the main package and 
        all player stuff will be in the player package.

I think the second way is  better because keeps all parts of the app that work together in the same package. Though of course PlayerFragment is technically hosted by MainActivity.

**Comment by Benji377** [comment]
Yeah, you are right, it really takes some time to find all the methods in a class you need. If we separate everything into three packages we might end up with three imports again. The second suggestion is much better since it's just one import for one class.
Is there a naming convention for this?

**Reply by squivix** [reply]
I'm sure there are conventions but I don't know. I just noticed other apps tend to have adapters in external classes and ui in a single package.

It's a tough transition but I think it's necessary, and it's good to do it early. I'll start working on this.
